nodejs/eks/cluster.ts

```diff
@@ -275,6 +275,10 @@ export class ClusterCreationRoleProvider extends pulumi.ComponentResource implem
     }
 }
 
+const getPartition = async() => {
+    return await (await aws.getPartition()).partition
```

Comment:
I would assume the `getPartition()` function doesn't cache the result internally, so we should probably cache it in an appropriate place in the call sites.

Reply:
TODO: Review if the getPartition method can be used with an explicit provider. Ideally, we should always honour the provider that's being used in the context of the component.

Reply:
@danielrbradley Can you point me to an example in the code base of a cached value? 

I'm not following on the explicit provider comments. I can certainly add the optional params to the function and pass that thru to the options object on the call to the aws lib. 

I actually think that this is more of a utility function than anything else. Basically where it matters, when building ARNs, we must be aware of the partition of the context on the run itself. I think this should just happen transparently for everyone using this lib.